src/parquet/column/reader.h

```diff
@@ -184,9 +188,9 @@ inline int64_t TypedColumnReader<DType>::ReadBatch(int batch_size, int16_t* def_
   }
 
   // Not present for non-repeated fields
-  if (descr_->max_repetition_level() > 0) {
+  if (descr_->max_repetition_level() > 0 && rep_levels) {
     num_rep_levels = ReadRepetitionLevels(batch_size, rep_levels);
-    if (num_def_levels != num_rep_levels) {
+    if (def_levels && num_def_levels != num_rep_levels) {
```

Comment:
The comment says `Set def_levels or rep_levels to nullptr if you want to skip reading them.` If the API user just wants to read `rep_levels` and not the `def_levels`, then this additional check is required. Alternatively, I feel we should throw a meaningful error message for improper use of API as well.